lib/bolt/transport/winrm/connection.rb

```diff
@@ -50,6 +50,7 @@ def connect
                       retry_limit: 1,
                       transport: transport,
                       ca_trust_path: cacert,
+                      basic_auth_only: target.options['basic-auth-only'],
```

Comment:
In order for this to use `basic` authentication we'll also need to set the `transport = :plaintext` when this option is specified https://github.com/WinRb/WinRM#plaintext

Reply:
I don't think this is right. I can use Basic Auth and do this using SSL as well. I configured our winrm server to only allow encrypted traffic. With the hacked bolt setting basic_auth_only to true, the connection works. So I would expect that it works as well with ssl transport. But I have only basic ruby knowledge and thus cannot deduce this from the code: https://github.com/WinRb/WinRM/blob/2a9a2ff55c5bbd903a019d63b1d134ac32ead4c7/lib/winrm/http/transport.rb#L263

Reply:
Ah, ok, yeah. I was thinking this could be used for both SSL and non-SSL transports, but I'm fine with limiting it to use with SSL to avoid plaintext until someone asks for it.